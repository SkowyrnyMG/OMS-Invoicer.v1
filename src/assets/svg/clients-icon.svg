<svg viewBox="0 0 24 16" xmlns="http://www.w3.org/2000/svg">
<path d="M23.9973 16.0009H8.00311L8.00044 15.1702C8.00044 13.4914 8.13378 12.522 10.1179 12.0633C12.3634 11.5446 14.5768 11.0819 13.5114 9.11784C10.3566 3.30018 12.6127 0 15.9996 0C19.3211 0 21.6345 3.17751 18.4877 9.11917C17.4543 11.0713 19.5891 11.5353 21.8812 12.0647C23.868 12.5234 24 13.4941 24 15.1755L23.9973 16.0009ZM0.00266681 16.0009H6.00433C5.97633 9.45919 8.99917 12.3047 8.99917 7.09906C8.99917 5.08962 7.68843 4.00022 6.00167 4.00022C3.50953 4.00022 1.7761 6.38302 4.13623 10.8393C4.91094 12.3047 3.31085 12.6514 1.59075 13.0487C0.0986722 13.3927 0 14.1221 0 15.3835L0.00266681 16.0009Z"/>
</svg>
