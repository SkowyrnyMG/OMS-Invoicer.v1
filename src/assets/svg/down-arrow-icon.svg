<svg width="20" height="20" viewBox="0 0 20 20" fill="none" xmlns="http://www.w3.org/2000/svg">
<rect x="0.5" y="0.5" width="18.33" height="18.33" rx="1.5" fill="#0078D7" stroke="black"/>
<rect x="8.625" y="3" width="1.75" height="10.5" fill="white"/>
<path d="M9.5 17L12.5311 13.0625H6.46891L9.5 17Z" fill="white"/>
</svg>
