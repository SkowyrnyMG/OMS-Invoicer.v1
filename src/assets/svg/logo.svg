<svg  viewBox="0 0 513 513" fill="none" xmlns="http://www.w3.org/2000/svg">
<path fill-rule="evenodd" clip-rule="evenodd" d="M133.138 1L265.276 133.228L133.138 265.455L1 133.228V1H133.138Z" fill="#F25C54"/>
<path fill-rule="evenodd" clip-rule="evenodd" d="M512.647 380.678V512.997H380.6L248.829 381.227L381.012 249.044L512.647 380.678Z" fill="#446DF6"/>
<path fill-rule="evenodd" clip-rule="evenodd" d="M381.058 1L248.829 133.228L381.012 265.411L512.647 133.777V1H381.058Z" fill="#446DF6"/>
<path fill-rule="evenodd" clip-rule="evenodd" d="M133.595 512.997H1V381.226L133.138 249L265.276 381.227L133.595 512.997Z" fill="#446DF6"/>
<rect width="362.04" height="362.04" rx="50" transform="matrix(0.707109 -0.707105 0.707109 0.707105 0.820312 256.999)" fill="#F8F8FF"/>
<path d="M146.679 301.499C137.595 301.499 129.387 299.541 122.054 295.624C114.804 291.707 109.095 286.332 104.929 279.499C100.845 272.582 98.8037 264.832 98.8037 256.249C98.8037 247.666 100.845 239.957 104.929 233.124C109.095 226.207 114.804 220.791 122.054 216.874C129.387 212.957 137.595 210.999 146.679 210.999C155.762 210.999 163.929 212.957 171.179 216.874C178.429 220.791 184.137 226.207 188.304 233.124C192.47 239.957 194.554 247.666 194.554 256.249C194.554 264.832 192.47 272.582 188.304 279.499C184.137 286.332 178.429 291.707 171.179 295.624C163.929 299.541 155.762 301.499 146.679 301.499ZM146.679 284.249C151.845 284.249 156.512 283.082 160.679 280.749C164.845 278.332 168.095 274.999 170.429 270.749C172.845 266.499 174.054 261.666 174.054 256.249C174.054 250.832 172.845 245.999 170.429 241.749C168.095 237.499 164.845 234.207 160.679 231.874C156.512 229.457 151.845 228.249 146.679 228.249C141.512 228.249 136.845 229.457 132.679 231.874C128.512 234.207 125.22 237.499 122.804 241.749C120.47 245.999 119.304 250.832 119.304 256.249C119.304 261.666 120.47 266.499 122.804 270.749C125.22 274.999 128.512 278.332 132.679 280.749C136.845 283.082 141.512 284.249 146.679 284.249Z" fill="black"/>
<path d="M293.853 299.999L293.728 247.499L267.978 290.749H258.853L233.228 248.624V299.999H214.228V212.499H230.978L263.728 266.874L295.978 212.499H312.603L312.853 299.999H293.853Z" fill="black"/>
<path d="M374.887 301.499C367.97 301.499 361.262 300.582 354.762 298.749C348.345 296.832 343.178 294.374 339.262 291.374L346.137 276.124C349.887 278.874 354.345 281.082 359.512 282.749C364.678 284.416 369.845 285.249 375.012 285.249C380.762 285.249 385.012 284.416 387.762 282.749C390.512 280.999 391.887 278.707 391.887 275.874C391.887 273.791 391.053 272.082 389.387 270.749C387.803 269.332 385.72 268.207 383.137 267.374C380.637 266.541 377.22 265.624 372.887 264.624C366.22 263.041 360.762 261.457 356.512 259.874C352.262 258.291 348.595 255.749 345.512 252.249C342.512 248.749 341.012 244.082 341.012 238.249C341.012 233.166 342.387 228.582 345.137 224.499C347.887 220.332 352.012 217.041 357.512 214.624C363.095 212.207 369.887 210.999 377.887 210.999C383.47 210.999 388.928 211.666 394.262 212.999C399.595 214.332 404.262 216.249 408.262 218.749L402.012 234.124C393.928 229.541 385.845 227.249 377.762 227.249C372.095 227.249 367.887 228.166 365.137 229.999C362.47 231.832 361.137 234.249 361.137 237.249C361.137 240.249 362.678 242.499 365.762 243.999C368.928 245.416 373.72 246.832 380.137 248.249C386.803 249.832 392.262 251.416 396.512 252.999C400.762 254.582 404.387 257.082 407.387 260.499C410.47 263.916 412.012 268.541 412.012 274.374C412.012 279.374 410.595 283.957 407.762 288.124C405.012 292.207 400.845 295.457 395.262 297.874C389.678 300.291 382.887 301.499 374.887 301.499Z" fill="black"/>
</svg>
