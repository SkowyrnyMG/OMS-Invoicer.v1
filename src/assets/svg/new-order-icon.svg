<svg  viewBox="0 0 40 31" fill="none" xmlns="http://www.w3.org/2000/svg">
<path fill-rule="evenodd" clip-rule="evenodd" d="M33.235 19.941H40V23.2645H33.235V30.0295H29.9115V23.2645H23.1465V19.941H29.9115V13.176H33.235V19.941ZM19.941 29.9115H0V26.588H19.941V29.9115ZM19.941 23.2246H0V19.9011H19.941V23.2246ZM19.941 16.6175H0V13.294H19.941V16.6175ZM39.882 9.9705H0V6.647H39.882V9.9705ZM39.882 3.3235H0V0H39.882V3.3235Z"/>
</svg>
