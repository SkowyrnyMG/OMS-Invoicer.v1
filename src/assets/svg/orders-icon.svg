<svg width="24" height="18" viewBox="0 0 24 18" fill="none" xmlns="http://www.w3.org/2000/svg">
<path fill-rule="evenodd" clip-rule="evenodd" d="M24 18H0V16H24V18ZM24 13.976H0V11.976H24V13.976ZM24 10H0V8H24V10ZM24 6H0V4H24V6ZM24 0V2H0V0H24Z" fill="black"/>
</svg>
