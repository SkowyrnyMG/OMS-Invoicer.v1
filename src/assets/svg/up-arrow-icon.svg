<svg width="20" height="20" viewBox="0 0 20 20" fill="none" xmlns="http://www.w3.org/2000/svg">
<rect x="18.8301" y="18.83" width="18.33" height="18.33" rx="1.5" transform="rotate(-180 18.8301 18.83)" fill="#0078D7" stroke="black"/>
<rect x="10.7051" y="16.33" width="1.75" height="10.5" transform="rotate(-180 10.7051 16.33)" fill="white"/>
<path d="M9.83008 2.33L6.79899 6.2675L12.8612 6.2675L9.83008 2.33Z" fill="white"/>
</svg>
