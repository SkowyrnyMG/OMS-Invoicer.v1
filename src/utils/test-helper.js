import React from 'react';
import { Router } from 'react-router-dom';
import { render, fireEvent } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { createMemoryHistory } from 'history';
import { theme } from 'themes/theme';

export const TestThemeProvider = ({ children }) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
);

const renderWithRouter = (ui, { route = '/', ...renderOptions } = {}) => {
  const history = createMemoryHistory({ initialEntries: [route] });
  const utils = render(<Router history={history}>{ui}</Router>, renderOptions);
  return {
    ...utils,
    history,
  };
};

const leftClick = { button: 0 };

export { leftClick, renderWithRouter, fireEvent };
