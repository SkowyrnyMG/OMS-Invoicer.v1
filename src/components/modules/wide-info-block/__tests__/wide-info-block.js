import React from 'react';

import { renderWithReduxRouter, screen } from 'utils/tests/test-helper';
import { ReactComponent as testIcon } from 'assets/svg/help-icon.svg';
import WideInfoBlock from '../wide-info-block';

describe('WideInfoBlock', () => {
  test('should display in the document with correct styles', () => {
    const { container } = renderWithReduxRouter(
      <WideInfoBlock Icon={testIcon} counter={2} title='test-title'>
        test-body-text
      </WideInfoBlock>,
    );

    const WideInfoBlockNode = container.firstChild;

    expect(WideInfoBlockNode).toBeInTheDocument();
    expect(WideInfoBlockNode).toMatchSnapshot();
  });

  test('should display image after receiving image prop', () => {
    renderWithReduxRouter(
      <WideInfoBlock
        image='https://test.pl'
        Icon={testIcon}
        counter={2}
        title='test-title'
      >
        test-body-text
      </WideInfoBlock>,
    );

    const imageNode = screen.getByRole('img', { name: 'position image' });

    expect(imageNode).toBeInTheDocument();
  });
});
