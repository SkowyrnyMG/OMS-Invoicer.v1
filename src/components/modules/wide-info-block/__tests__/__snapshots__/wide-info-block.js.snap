// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WideInfoBlock should display in the document with correct styles 1`] = `
.c0 {
  padding: 5rem 0;
  display: grid;
  grid-template-columns: 5rem [start] repeat(2,1fr) [end] 5rem;
  grid-column: 1 / -1 !important;
  grid-gap: 3%;
  grid-row-gap: 5%;
  min-height: 60rem;
  background-color: #F8F8FF;
  margin: 0;
  padding-top: 10rem;
}

.c0 > * {
  grid-column: start / end;
}

.c1 {
  display: grid;
  grid-template-columns: repeat(2,1fr);
  grid-column-gap: 10rem;
  text-align: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  position: relative;
  margin: 0;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  font-size: 34px;
}

.c4 {
  position: absolute;
  line-height: 1;
  top: 0;
  left: 0;
  font-size: 180px;
  color: #446DF644;
  -webkit-transform: translate(-100%,-50%);
  -ms-transform: translate(-100%,-50%);
  transform: translate(-100%,-50%);
  z-index: 0;
}

.c4 svg {
  position: absolute;
  width: 70px;
  fill: #446DF6;
  left: 0;
  top: 50%;
  -webkit-transform: translate(-50%,-30%);
  -ms-transform: translate(-50%,-30%);
  transform: translate(-50%,-30%);
}

.c5 {
  text-align: left;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  padding-left: calc(15rem + 3%);
}

@media (max-width:680px) {
  .c0 {
    grid-template-columns: [start] 1fr [end];
    padding: 2rem;
  }
}

@media (max-width:1028px) {
  .c1 {
    grid-template-columns: 1fr;
    grid-column-gap: 0;
  }
}

@media (max-width:680px) {
  .c1 {
    position: relative;
    padding-top: 15rem;
  }
}

@media (max-width:680px) {
  .c3 {
    position: static;
  }
}

@media (max-width:680px) {
  .c4 {
    font-size: 180px;
    left: 50%;
    -webkit-transform: translate(-50%,0);
    -ms-transform: translate(-50%,0);
    transform: translate(-50%,0);
  }
}

@media (max-width:680px) {
  .c2 {
    padding-left: 0;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <h3
        class="c3"
      >
        <div
          class="c4"
        >
          <svg>
            help-icon.svg
          </svg>
          2
        </div>
        test-title
      </h3>
      
    </div>
    <div
      class="c5"
    >
      test-body-text
    </div>
  </div>
</div>
`;
